crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -1220,6 +1249,28 @@ where
         &self,
         revert_until: BlockNumber,
     ) -> Result<Option<Chain>, CanonicalError> {
+        if self
+            .externals
+            .provider_factory
+            .static_file_provider()
+            .get_highest_static_file_block(StaticFileSegment::Headers)
+            .unwrap_or_default() >
```

Comment:
can we move parts of the docs above the if check to explain why we're checking this